app/src/main/java/com/codesoom/assignment/application/auth/AuthenticationService.java

```diff
@@ -0,0 +1,40 @@
+package com.codesoom.assignment.application.auth;
+
+import com.codesoom.assignment.application.users.UserNotFoundException;
+import com.codesoom.assignment.domain.users.User;
+import com.codesoom.assignment.domain.users.UserRepository;
+import com.codesoom.assignment.dto.LoginRequestDto;
+import com.codesoom.assignment.utils.JwtUtil;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationService {
+
+    private final JwtUtil jwtUtil;
+    private final UserRepository repository;
+
+    public AuthenticationService(JwtUtil jwtUtil, UserRepository repository) {
+        this.jwtUtil = jwtUtil;
+        this.repository = repository;
+    }
+
+    /**
+     * 로그인 정보로 인증하고, 토큰을 발급합니다.
+     *
+     * @param loginRequestDto 로그인 요청 정보
+     * @return 발급된 토큰
+     * @throws UserNotFoundException 이메일과 일치하는 회원이 없는 경우
+     * @throws InvalidPasswordException 비밀번호가 일치하지 않을 경우
+     */
+    public String login(LoginRequestDto loginRequestDto) {
+        User user = repository.findByEmail(loginRequestDto.getEmail())
+                .orElseThrow(() -> new UserNotFoundException("이메일과 일치하는 회원을 찾을 수 없습니다."));
```

Comment:
예외 메시지를 작성할 때 이렇게 해 보세요.

"~때문에 ~을 할 수 없습니다."

```suggestion
                .orElseThrow(() -> new UserNotFoundException("회원을 찾을 수 없으므로 로그인에 실패했습니다."));
```

로그를 읽게 되는 것은 새벽 3시에 디버깅하는 미래의 자기 자신이라 상상하세요. 원인만 설명해서는 디버깅할 때 힌트가 부족한 일이 벌어져서 10분이면 끝날 일이 몇시간 걸리기도 합니다.

Reply:
예외 메시지 작성 시 원인과 결과를 꼭 포함하는 습관을 가져야겠습니다. 💪